src/string/grapheme.cr

```diff
@@ -0,0 +1,19 @@
+require "./grapheme/grapheme"
+
+class String
+  # returns graphemes cluster iterator over Unicode extended grapheme clusters, specified in the Unicode Standard Annex #29. Grapheme clusters correspond to
+  # "user-perceived characters". These characters often consist of multiple code points (e.g. the "woman kissing woman" emoji consists of 8 code points:
+  # woman + ZWJ + heavy black heart (2 code points) + ZWJ + kiss mark + ZWJ + woman) and the rules described in Annex #29 must be applied to group those
+  # code points into clusters perceived by the user as one character.
+  # ```
+  # "🔮👍🏼!".each do |cluster|
+  #   pp cluster.codepoints
+  #   pp cluster.positions
+  #   pp cluster.str
+  #   pp cluster.bytes
+  # end
+  # ```
+  def graphemes
```

Comment:
I think this should return `Array(Grapheme::Cluster)`. Then we can have methods like `each_grapheme` that iterates over the graphemes or return an `Iterator` if there's no block. That would make it consistent with other String APIs.

Reply:
Thank you Ary. Will make changes

Reply:
Made changes. Added `each_grapheme` methods as well